docs/admin/templates/extending-templates/workspace-startup-coordination.md

```diff
@@ -0,0 +1,484 @@
+# Startup Dependencies in Templates
+
+When workspaces start, scripts often need to run in a specific order.
+For example, an IDE or coding agent might need the repository cloned
+before it can start. Without explicit coordination, these scripts can
+race against each other, leading to startup failures and inconsistent
+workspace states.
+
+Coder's workspace startup coordination feature allows scripts to declare
+dependencies on each other, ensuring they execute in the correct order. This
+eliminates race conditions and makes workspace startup predictable and
+reliable.
+
+> [!NOTE]
+> This feature is experimental and available under the `coder exp sync`
+> command namespace.
+
+## Benefits
+
+- **Eliminates race conditions** during workspace startup
+- **Explicit ordering** makes initialization predictable and debuggable
+
+## What is Startup Coordination?
+
+Startup coordination is built around the concept of **units**. A unit is a
+named phase of work, typically corresponding to a script or initialization
+task. Units can declare dependencies on other units, creating an explicit
+ordering for workspace initialization.
+
+**Common use case:**
+
+- Repository must be available before starting development tools
```

Comment:
Honestly, I expected a longer paragraph here. Perhaps you should combine it with other?

Reply:
Updated